internal/dag/gatewayapi_processor.go

```diff
@@ -71,8 +71,23 @@ func (p *GatewayAPIProcessor) Run(dag *DAG, source *KubernetesCache) {
 		var matchingRoutes []*gatewayapi_v1alpha1.HTTPRoute
 		var listenerSecret *Secret
 
+		// Validate the Kind on the selector is a supported type.
+		if listener.Protocol != gatewayapi_v1alpha1.HTTPProtocolType &&
+			listener.Protocol != gatewayapi_v1alpha1.HTTPSProtocolType &&
+			listener.Protocol != gatewayapi_v1alpha1.TLSProtocolType {
+			p.Errorf("Listener.Protocol %q is not supported.", listener.Protocol)
+			continue
+		}
+
 		// Check for TLS on the Gateway.
 		if listener.TLS != nil {
+			if listener.Protocol != gatewayapi_v1alpha1.HTTPSProtocolType &&
```

Comment:
only point of contention for this PR might be if this is the right choice

https://github.com/kubernetes-sigs/gateway-api/blob/e3483c5bdf32f2468ca3c17cf29a988808c132fd/apis/v1alpha1/gateway_types.go#L187-L189 says "ignored otherwise" which could be interpreted as the opposite of this

Reply:
Yeah that does sound like we should still process routes for the listener.

Reply:
👍